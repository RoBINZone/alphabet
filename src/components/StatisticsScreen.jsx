import React, { useState } from "react";
import { StatisticsService } from "../services/statistics-service.js";
import { SET_SCREEN, WELCOME_SCREEN } from "../consts.js";

export const StatisticsScreen = ({ dispatch }) => {
  const [statistics, setStatistics] = useState(
    StatisticsService.getStatistic(),
  );

  return (
    <div className="statistics-screen">
      <div> дата / правильно / неправильно / середній час відповіді</div>
      {statistics &&
        statistics.reverse().map((item, index) => (
          <div className="statistics-list-item" key={index}>
            <div className="statistics-list-item-cell datetime">
              {StatisticsService.formatDate(item.datetime)}
            </div>
            <div className="statistics-list-item-cell correct">
              {item.correct}
            </div>
            <div className="statistics-list-item-cell incorrect">
              {item.incorrect}
            </div>
            <div className="statistics-list-item-cell average-response-time">
              {item.averageResponseTime} c.
            </div>
          </div>
        ))}
      <div>
        <button
          onClick={() => dispatch({ type: SET_SCREEN, screen: WELCOME_SCREEN })}
        >
          На початок
        </button>
        <button
          onClick={() => {
            StatisticsService.clearStatistics();
            setStatistics([]);
          }}
        >
          Очистити статистику
        </button>
      </div>
    </div>
  );
};
