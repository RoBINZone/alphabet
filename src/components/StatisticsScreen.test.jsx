import React from "react";
import { render } from "@testing-library/react";
import { expect, it, describe } from "vitest";
import { StatisticsScreen } from "./StatisticsScreen.jsx";

describe("Statistics Screen", () => {
  it("should match snapshot", () => {
    localStorage.setItem(
      "statistic",
      JSON.stringify([
        {
          datetime: 1749905469999,
          correct: 10,
          incorrect: 10,
          averageResponseTime: 0.24,
        },
      ]),
    );
    const { asFragment } = render(
      <StatisticsScreen onScreenChange={() => undefined} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

[
  {
    datetime: 1749905469999,
    correct: 10,
    incorrect: 10,
    averageResponseTime: 0.24,
  },
];
