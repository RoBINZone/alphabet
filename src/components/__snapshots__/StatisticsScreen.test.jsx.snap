// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Statistics Screen > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="statistics-screen"
  >
    <div>
       дата / правильно / неправильно / середній час відповіді
    </div>
    <div
      class="statistics-list-item"
    >
      <div
        class="statistics-list-item-cell datetime"
      >
        6/14/2025, 3:51:09 PM
      </div>
      <div
        class="statistics-list-item-cell correct"
      >
        10
      </div>
      <div
        class="statistics-list-item-cell incorrect"
      >
        10
      </div>
      <div
        class="statistics-list-item-cell average-response-time"
      >
        0.24 c.
      </div>
    </div>
    <div>
      <button>
        На початок
      </button>
      <button>
        Очистити статистику
      </button>
    </div>
  </div>
</DocumentFragment>
`;
